pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/EasyPostUtil.java

```diff
@@ -307,45 +309,29 @@ public Shipment getShipment(String shipmentId) throws EasyPostException {
      * @return true if easypost was able to create the address, false otherwise
      */
 
-    public boolean checkAddress(@NonNull JuniperKitRequest juniperKitRequest, String phone) {
-        if ((StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID()))
-                || StringUtils.isBlank(juniperKitRequest.getLastName())) {
-            return false;
+    public String getEasyPostAddressId(@NonNull JuniperKitRequest juniperKitRequest, String phone, DeliveryAddress deliveryAddress)
+            throws DsmInternalError, DSMBadRequestException {
+        if (StringUtils.isBlank(juniperKitRequest.getLastName())) {
+            throw new DSMBadRequestException("KitRequest did not have a last name ");
         }
-        //let's validate the participant's address
-        String name = "";
-        if (StringUtils.isNotBlank(juniperKitRequest.getFirstName())) {
-            name += juniperKitRequest.getFirstName() + " ";
-        }
-        name += juniperKitRequest.getLastName();
         if (juniperKitRequest.isSkipAddressValidation()) {
             //if no validation is needed, we just need to create the Address instance in easypost and get its id back
             try {
                 Address address = createAddressWithoutValidation(name, juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(),
                         juniperKitRequest.getCity(),
                         juniperKitRequest.getPostalCode(), juniperKitRequest.getState(), juniperKitRequest.getCountry(), phone);
-                juniperKitRequest.setEasypostAddressId(address.getId());
-                return true;
+                return address.getId();
             } catch (EasyPostException e) {
                 // log the reason for address creation failure and return false. The method will then return the error code
-                logger.warn("Easypost couldn't create an address for " + juniperKitRequest.getShortId(), e);
-                return false;
+                throw new DsmInternalError("Easypost couldn't create an address for " + juniperKitRequest.getShortId(), e);
             }
         }
-        //to validate the address, first we need create the Address instance
-        DeliveryAddress deliveryAddress =
-                new DeliveryAddress(juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(), juniperKitRequest.getCity(),
-                        juniperKitRequest.getState(),
-                        juniperKitRequest.getPostalCode(), juniperKitRequest.getCountry(), name, phone);
```

Comment:
moved to NonPepperKitCreationService. Now DeliverAddress instance is calculated first and then passed to the easypost util